projects/plugins/jetpack/tools/webpack.config.search.js

```diff
@@ -7,11 +7,28 @@ const {
 	defaultRequestToExternal,
 	defaultRequestToHandle,
 } = require( '@wordpress/dependency-extraction-webpack-plugin/util' );
+const { readFileSync } = require( 'fs' );
 const path = require( 'path' );
 const webpack = require( 'webpack' );
 
 const isDevelopment = process.env.NODE_ENV !== 'production';
 
+/**
+ * As the version number in package.json deesn't have the same format as it is in PHP.
+ * So we extract the Jetpack version from plugin PHP file and to make sure Webpack
+ * gets exactly the same.
+ *
+ * Examples of JETPACK__VERSION are: `9.8-alpha`, `9.8-beta`, `9.8`, `9.8.1`, `9.8.2`.
+ */
+const pluginFileContent = readFileSync( path.join( __dirname, '../jetpack.php' ) ).toString();
+const jetpackAlphaNumericVersionMatch = pluginFileContent.match(
+	/define\(\s+'JETPACK__VERSION',\s+'([0-9A-Za-z.-]+)'\s+\)/
+);
+const bundleVersionExtension =
+	jetpackAlphaNumericVersionMatch?.length > 0
+		? `.${ jetpackAlphaNumericVersionMatch[ 1 ].replace( /[^0-9A-Za-z]+/g, '-' ) }`
+		: '';
```

Comment:
As currently implemented, this version extraction logic can silently fail. Is this desirable?

Reply:
Yep if no JETPACK__VERSION found, then it'll build the bundle name without the version. 

If it fails to get a version, then it's wpcom.
If it get the version, then it's the plugin.

Hmmm, possibly not really very ideal and might due to change. Any suggestion on this?

Reply:
It might be better to use the modified timestamp as the fallback instead of a blank string.